.github/workflows/ci.yml

```diff
@@ -0,0 +1,30 @@
+name: ci
+
+on: [pull_request, push]
+
+jobs:
+  docker:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+      - name: Build
+        uses: docker/build-push-action@v3
+
+  sonar:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: recursive
+      - name: Set up JDK
+        uses: actions/setup-java@v3
+        with:
+          java-version: "11"
+          distribution: "temurin"
+      - name: Run Sonar
+        env:
+          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+        run: |

```

Comment:
Unit tests are not being run so no chance for coverage report to be generated. The README says `mvn test` is supported, need to incorporate that into CI and get it to output a coverage report for sonar to import.